scripts/databricks_install.py

```diff
@@ -0,0 +1,272 @@
+      """,
+        formatter_class=argparse.ArgumentDefaultsHelpFormatter,
+    )
+    parser.add_argument(
+        "--profile",
+        help="The CLI profile to use for connecting to the databricks workspace",
+        default="DEFAULT",
+    )
+    parser.add_argument(
+        "--path-to-recommenders",
+        help="The path to the root of the recommenders repository. Default assumes that the script is run in the root of the repository",
+        default=".",
+    )
+    parser.add_argument(
+        "--eggname",
+        help="Name of the egg you want to generate. Useful if you want to name based on branch or date.",
+        default="Recommenders.egg",
+    )
+    parser.add_argument(
+        "--dbfs-path",
+        help="The directory on dbfs that want to place files in",
+        default="dbfs:/FileStore/jars",
+    )
+    parser.add_argument(
+        "--overwrite",
+        action="store_true",
+        help="Whether to overwrite existing files.",
+        default=False,
+    )
+    parser.add_argument(
+        "--prepare-o16n",
+        action="store_true",
+        help="Whether to install additional libraries for operationalization.",
+        default=False,
+    )
+    parser.add_argument(
+        "--mmlspark",
+        action="store_true",
+        help="Whether to install mmlspark.",
+        default=False,
+    )
+    parser.add_argument("cluster_id", help="cluster id for the cluster to install on.")
+    args = parser.parse_args()
+
+    ###############################
+    ## Create the egg:
+    ###############################
+
+    print("Preparing Recommenders library file ({})...".format(args.eggname))
+    myegg = create_egg(
+        args.path_to_recommenders, local_eggname=args.eggname, overwrite=args.overwrite
+    )
+    print("Created: {}".format(myegg))
+
+    ###############################
+    ## Interact with Databricks:
+    ###############################
+
+    ## first make sure you are using the correct profile and connecting to the intended workspace
+    my_api_client = _get_api_client(ProfileConfigProvider(args.profile).get_config())
+
+    ## upload the egg:
+    uploadpath = "/".join([args.dbfs_path, os.path.basename(myegg)])
+    print(
+        "Uploading {} to databricks at {}".format(os.path.basename(myegg), uploadpath)
+    )
+    DbfsApi(my_api_client).cp(
+        recursive=False, src=myegg, dst=uploadpath, overwrite=args.overwrite
+    )
+
+    ## steps below require the cluster to be running. Check status
+    try:
+        status = ClusterApi(my_api_client).get_cluster(args.cluster_id)
+    except HTTPError as e:
+        print(e)
+        print(textwrap.dedent(CLUSTER_NOT_FOUND_MSG.format(args.cluster_id)))
+        raise
+
+    if status["state"] != "RUNNING":
+        print(
+            textwrap.dedent(
+                CLUSTER_NOT_RUNNING_MSG.format(args.cluster_id, status["state"])
+            )
+        )
+        sys.exit()
```

Comment:
Can we force-start cluster here, wait until done, and then install egg instead of exit? At this moment, a user's intention is to install the repo anyway.

Reply:
i did a version of this with pending status, we could generalize and just make it the default behavior... The issue is just the waiting, but I suppose they'd be waiting no matter whether we do it for them or not.